src/libstd/path.rs

```diff
@@ -1859,6 +1860,27 @@ impl Path {
     pub fn is_dir(&self) -> bool {
         fs::metadata(self).map(|m| m.is_dir()).unwrap_or(false)
     }
+
+    /// Checks if the path buffer is empty. On Windows, it will return false if the inner string is
+    /// invalid unicode. On Unix, this is a no-op.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::path::Path;
+    ///
+    /// let path = Path::new("/tmp/foo.rs");
+    ///
+    /// assert!(!path.is_empty());
+    /// ```
+    #[unstable(feature = "path_extras", reason = "recently added", issue = "30259")]
+    pub fn is_empty(&self) -> bool {
+        if let Some(b) = self.inner.to_bytes() {
```

Comment:
I think it's better to use `os_str_as_u8_slice` and check if it is empty. It's not fallible and it's simple.